small_text/query_strategies/coresets.py

```diff
@@ -161,37 +163,48 @@ def lightweight_coreset(x, x_mean, n, normalized=False, proba=None):
     """
     _check_coreset_size(x, n)
 
-    sim = x.dot(x_mean)
-    if not normalized:
-        sim = sim / (np.linalg.norm(x, axis=1) * np.linalg.norm(x_mean))
+    num_batches = int(np.ceil(x.shape[0] / batch_size))
+    indices = np.array([])
 
-    dists = np.arccos(sim) / np.pi
-    dists = np.square(dists)
+    for x_batch in np.array_split(x, num_batches, axis=0):
 
-    sum_dists = dists.sum()
+        sim = x_batch.dot(x_mean)
+        if not normalized:
+            sim = sim / (np.linalg.norm(x_batch, axis=1) * np.linalg.norm(x_mean))
 
-    if proba is None:
-        uniform = 0.5 * 1 / x.shape[0]
-        proba = uniform + 0.5 * dists / sum_dists
-    else:
-        proba = 0.5 * proba / proba.sum() + 0.5 * dists / sum_dists
+        dists = np.arccos(sim) / np.pi
+        dists = np.square(dists)
+
+        sum_dists = dists.sum()
 
-    proba = proba / np.linalg.norm(proba, ord=1)
+        if proba is None:
+            uniform = 0.5 * 1 / x_batch.shape[0]
+            proba = uniform + 0.5 * dists / sum_dists
+        else:
+            proba = 0.5 * proba / proba.sum() + 0.5 * dists / sum_dists
 
-    return np.random.choice(np.arange(x.shape[0]), n, replace=False, p=proba)
+        proba = proba / np.linalg.norm(proba, ord=1)
+
+        indices_batch = np.random.choice(np.arange(x_batch.shape[0]), n, replace=False, p=proba)
+        indices = np.concatenate((indices, indices_batch))
+
+    return indices.astype(np.int64)
```

Comment:
Minor nitpick: I used to do this as well, but I think this is a symptom of not intializing the numpy array correctly: instead of `indices.astype(np.int64)` I would initialize the array indices with `indices = np.array([], dtype=int)`.

Moreover, I would prefer int over np.int64 to make it consistent with the query startegies (which are admittedly just in the process of getting type annotations, so this is difficult to spot).

Reply:
Yeah, it makes sense. Using `np.int64` was a quick fix as I was failing unit tests. I have updated the code.